### ISSUE TITLE
False positive E1126 when using np.newaxis with np.where

### ISSUE THREAD

**Issue by SentinelCLS** [PyCQA/pylint#10312]
### Bug description

```python
from numpy import zeros, where, newaxis

omega = zeros(10)

w_1 = zeros((10, 5))
w_2 = where(omega < 1., 0., 1.)

result = w_1 * w_2[:, newaxis]
```

### Command used

```shell
python -m pylint test.py
```

### Pylint output

```python
E1126: Sequence index is not an int, slice, or instance with __index__ (invalid-sequence-index)
```

### Expected behavior

No error reported

### Pylint version

```shell
pylint 3.3.1
astroid 3.3.5
Python 3.10.14 | packaged by conda-forge | (main, Mar 20 2024, 12:45:18) [GCC 12.3.0]
```

### Additional dependencies

```python
numpy==1.24.4
```
